src/export/StructureDefinitionExporter.ts

```diff
@@ -131,13 +375,31 @@ export class StructureDefinitionExporter implements Fishable {
     delete structDef.keyword;
     // keep structDef.fhirVersion as that ought not change from parent to child
     // keep mapping since existing elements refer to the mapping and we're not removing those
-    // keep kind since it should not change
-    structDef.abstract = false; // always reset to false, assuming most children of abstracts aren't abstract
+    // keep kind since it should not change except for logical models
+    if (fshDefinition instanceof Logical) {
+      structDef.kind = 'logical';
+    }
+    structDef.abstract = false; // always reset to false, assuming most children of abstracts aren't abstract; can be overridden w/ rule
+    // context and contextInvariant only apply to extensions.
     // keep context, assuming context is still valid for child extensions
     // keep contextInvariant, assuming context is still valid for child extensions
-    // keep type since this should not change for profiles or extensions
+    if (!(fshDefinition instanceof Extension)) {
+      // Should not be defined for profiles, logical models, but clear just to be sure
+      delete structDef.context;
+      delete structDef.contextInvariant;
+    }
+    // keep type since this should not change except for logical models or resources
+    if (fshDefinition instanceof Logical || fshDefinition instanceof Resource) {
+      // By definition, the 'type' is the same as the 'id'
+      structDef.type = fshDefinition.id;
+    }
     structDef.baseDefinition = baseURL;
-    structDef.derivation = 'constraint'; // always constraint since SUSHI only supports profiles/extensions right now
+    if (fshDefinition instanceof Logical || fshDefinition instanceof Resource) {
+      structDef.derivation = 'specialization';
```

Comment:
Do we want to just combine these two `if` statements since they have the same predicate.

Reply:
Good catch... will combine